popde/density_estimate.py

```diff
@@ -120,6 +120,70 @@ def prepare_data(self):
         else:
             self.kde_data = self.std_data
 
+
+    def evaluate_with_transf(self, points):
+        """
+        Transforms the evaluate points based on specified transformations
+        , evaluates the kernel density estimate (KDE) on the transformed points
+         and adjusts the KDE values using the Jacobian of the transformations.
+
+        Parameters:
+        -----------
+        points : array-like, shape (n_samples, n_features)
+            The original parameter space points to be transformed and evaluated.
+
+        Returns:
+        --------
+        kde_vals : array-like, shape (n_samples,)
+        The KDE values adjusted by the Jacobian of the transformations.
+        """
+        ### Transform points with transf, normalize, rescale
+        if self.input_transf is not None:
+            transf_points = transf.transform_data(points, self.input_transf)
+        else:
+            transf_points = points
+        #normalize
+        if self.stdize:
+            std_transf = ['stdize' for dim in self.ndim]
+            std_values_for_points = np.std(transf_points, axis=0)  # record the stds
+            std_points = transf.transform_data(transf_points, std_transf)
+        else:
+            std_values_for_points = None
+            std_points = transf_points
+
+        #rescale
+        if self.rescale is not None:
+            transf_data = transf.transform_data(std_points, self.rescale)
+        else:
+            transf_data = std_points
+
+        #Evaluate kde on transform points
+        kde_vals = self.evaluate(transf_data)
+
+        num_points, num_dims = transf_data.shape
+        if len(self.input_transf) != num_dims:
+            raise ValueError("The transformation list length must match the number of dimensions in eval_vals")
```

Comment:
I am not seeing why we need this, or why it is correct .. self.input_transf can be None, and if it is not None we have the check_dimensionality method to determine if the length is the right one. In any case the evaluate method should fail if transf_data has the wrong number of dimensions.